### DISCUSSION TITLE
Syntax for Conditions by Flows

### DISCUSSION THREAD

**Discussion by rynskiy** [directus/directus#15568] [Archive]
Hi,

I need help with the next task:

- I have in flow a Read Data with next conditions (see screenshot) where i have a **filter** and **agregate** with **groupBy** "edition"
![Flow1](https://user-images.githubusercontent.com/54798086/190125363-fd2b3aef-83f5-4ee6-a13f-36073c7016dd.PNG)

- at the next **Condition** step, I need to set a simple condition: if count id is 5 -> send Notification MATCH. If count id isn't 5 -> send Notification NO MATCH. I have set this condition like: 
` {
    "$last[0].count": {
        "id": {
            "_eq": 5
        }
    }
}`

... and it doesn't work correctly. Which conditions must be used here?
![payload](https://user-images.githubusercontent.com/54798086/190126717-94a58abd-f13e-492a-ae43-22f98d3700ae.PNG)
![condition](https://user-images.githubusercontent.com/54798086/190127045-908304a6-e039-4cd4-993c-535e8a19eb8e.PNG)

**Comment by rynskiy** [comment]
Is anybody here? ;)

**Comment by lukas-schaetzle** [comment]
Looks like you simply forgot to put the variable in curly braces. Try `"{{$last[0].count}}"` instead of `"$last[0].count"` in the condition's filter object.

**Reply by danielband** [reply]
Nope, I just tried that. Somehow the condition operation does not play well with arrays.

**Comment by rynskiy** [comment]
Are there any updates about this problem/bug with condition operation and arrays by Flows? Actually, this problem with Conditions in Flows leaves us in doubt as to the advisability of using Directus in general...

**Reply by nickrum** [reply]
This might be a limitation of the current condition syntax. Have you tried using

```json
{
  "$last": {
    "0": {
      "count": {
        "id": {
          "_eq": 5
        }
      }
    }
  }
}
```

? If that doesn't work, you might have to use a Script Operation to check for the condition.

**Reply by domsen123** [reply]
JFYI: this syntax is not working -> you have to use script operation to either check the condition in script block or transform array data in script block and check transformed data in condition afterwards.

I tested realy every syntax in condition.... as soon as it is an array -> error... no chance to select an index in an array

**Reply by nickrum** [reply]
Thanks for confirming @domsen123 👍
The filter/condition syntax works quite well when used as a filter in conjunction with the API, but it has its shortcomings when used to assert certain conditions on an arbitrary json object, unfortunately.

**Comment by emilyf** [comment]
@rynskiy yes I think @domsen123 is correct here; he posted on discord this solution

<img width="1072" alt="Screen Shot 2022-10-24 at 6 05 28 AM" src="https://user-images.githubusercontent.com/1789807/197521945-48c37749-d876-470e-b396-e9ff36d349e8.png">

**Comment by SiteSeerBE** [comment]
Since I don't like to end on a throw and you might want to use the true and false conditions, might I suggest this solution:

Run Script
```
module.exports = async function(data) {
    const result = {{$last[0]}}
    if(result.count.id === 5) return {notify:true}
    return {notify:false};
} 
```
Condition
```
{
    "$last": {
        "notify": {
            "_eq": true
        }
    }
}
```

**Reply by Yves852** [reply]
Thank you very much. The condition block was driving me crazy without being able to figure out what where wrong reading the doc. 
Basically wanted a webhook flow from a contact form, validate it has a email and message then proceed the email. 

Splitting it into a script block and a condition block like your solution did the thing. 

```js
module.exports = async function(data) {
    const result = {{$last}}
   
    // Guards
    if(result.body.email === null || result.body.email === '') return {notify:false}
    if(result.body.message === null || result.body.message === '') return {notify:false}

    return {notify:true};
} 
```

**Reply by alilostab14-cmyk** [reply]
Hi everyone
I created a flow to control duplicate records, and I implemented the following configuration:

1. Read Data Operation:

{
    "filter": {
        "_and": [
            { "Ense": { "_eq": "{{ $trigger.payload.Ense }}" } },
            { "activite": { "_eq": "{{ $trigger.payload.activite }}" } },
            { "date_cours": { "_eq": "{{ $trigger.payload.date_cours }}" } }
        ]
    }
}
2. Add Script Operation:(check_dup)

module.exports = async function(data) {
    return (data.read_data.length > 0);
}
This script returns true if a duplicate exists.

3. Condition Operation:

{
    "check_dup": {
        "_eq": true
    }
}
The issue:

Even when there is no duplicate, Directus still returns this error:

The following fields contain invalid values:
check_dup: The value must be true
It seems that Directus is requiring the value true even when the condition should evaluate to false (i.e., no duplicates found).

Question:

How can I properly set up the flow so that Directus only triggers the error when a duplicate exists, and allows the record to be created when there is no duplicate?
thank in advance
